src/tests/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs

```diff
@@ -253,6 +262,271 @@ static bool CollectCrashDump(Process process, string path)
             return fSuccess && createdump.ExitCode == 0;
         }
 
+        private static List<string> knownNativeModules = new List<string>() { "libcoreclr.so", "libclrjit.so" };
+        private static string TO_BE_CONTINUE_TAG = "<TO_BE_CONTINUE>";
+        private static string SKIP_LINE_TAG = "# <SKIP_LINE>";
+
+
+        static bool RunProcess(string fileName, string arguments)
+        {
+            Process proc = new Process()
+            {
+                StartInfo = new ProcessStartInfo()
+                {
+                    FileName = fileName,
+                    Arguments = arguments,
+                    UseShellExecute = false,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                }
+            };
+
+            Console.WriteLine($"Invoking: {proc.StartInfo.FileName} {proc.StartInfo.Arguments}");
+            proc.Start();
+
+            Task<string> stdOut = proc.StandardOutput.ReadToEndAsync();
+            Task<string> stdErr = proc.StandardError.ReadToEndAsync();
+            if(!proc.WaitForExit(DEFAULT_TIMEOUT_MS))
+            {
+                proc.Kill(true);
+                Console.WriteLine($"Timedout: '{fileName} {arguments}");
+                return false;
+            }
+
+            Task.WaitAll(stdOut, stdErr);
+            string output = stdOut.Result;
+            string error = stdErr.Result;
+            if (!string.IsNullOrWhiteSpace(output))
+            {
+                Console.WriteLine($"stdout: {output}");
+            }
+            if (!string.IsNullOrWhiteSpace(error))
+            {
+                Console.WriteLine($"stderr: {error}");
+            }
+            return true;
+        }
+
+        /// <summary>
+        ///     Parse crashreport.json file, use llvm-symbolizer to extract symbols
+        ///     and recreate the stacktrace that is printed on the console.
+        /// </summary>
+        /// <param name="crashReportJsonFile">crash dump path</param>
+        /// <param name="outputWriter">Stream for writing logs</param>
+        /// <returns>true, if we can print the stack trace, otherwise false.</returns>
+        static bool TryPrintStackTraceFromCrashReport(string crashReportJsonFile, StreamWriter outputWriter)
+        {
+            if (OperatingSystem.IsLinux() || OperatingSystem.IsMacOS())
+            {
+                if (!RunProcess("sudo", $"ls -l {crashReportJsonFile}"))
+                {
+                    return false;
+                }
+
+                Console.WriteLine("=========================================");
+                string userName = Environment.GetEnvironmentVariable("USER");
+                if (!string.IsNullOrEmpty(userName))
+                {
+                    if (!RunProcess("sudo", $"chown {userName} {crashReportJsonFile}"))
```

Comment:
If elevating works on Mac right now, that's great, but I could see this not working forever, so we should give a little thought to what we would do instead.